### PR TITLE
Document and test the diskQueueACKs helper in the disk queue

### PR DESCRIPTION
**Pull request by faec**
## What does this PR do?

Add many comments to `acks.go`, unit test `diskQueueACKs`, and fix corner cases in `diskQueueACKs.addFrames`.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: libbeat/publisher/queue/diskqueue/acks_test.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	testCases := []struct {
+		name     string
+		frameID  frameID
+		position queuePosition
+		steps    []addFramesTestStep
+	}{
```
Comment (urso) [inline libbeat/publisher/queue/diskqueue/acks_test.go:47]: nit; `cases := map[string]struct{ ... }{ "2-segment test": ... }`

That way the test execution is also randomized, ensuring that we don't have hidden dependencies on state in between tests in general.

File: libbeat/publisher/queue/diskqueue/acks_test.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+			},
+		},
+	}
+
+	path := filepath.Join(dir, "state.dat")
+	stateFile, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer stateFile.Close()
```
Comment (urso) [inline libbeat/publisher/queue/diskqueue/acks_test.go:233]: why share the state file between tests? Does that create a dependency between tests? Consider to have a separate directory per test, so states and contents is completely independent.

File: libbeat/publisher/queue/diskqueue/acks_test.go (hunk 1 above)
Comment (faec) [reply libbeat/publisher/queue/diskqueue/acks_test.go]: Done

File: libbeat/publisher/queue/diskqueue/acks_test.go (hunk 2 above)
Comment (faec) [reply libbeat/publisher/queue/diskqueue/acks_test.go:252]: Done

**Review by urso — APPROVED** [review]
